readthedocs/search/parse_json.py

```diff
@@ -59,40 +59,44 @@ def generate_sections_from_pyquery(body):
         }
 
 
-def process_file(filename):
-    """Read a file from disk and parse it into a structured dict."""
+def process_file(fjson_filename, filename):
+    """Read the fjson file from disk and parse it into a structured dict."""
     try:
-        with codecs.open(filename, encoding='utf-8', mode='r') as f:
+        with codecs.open(fjson_filename, encoding='utf-8', mode='r') as f:
             file_contents = f.read()
     except IOError:
-        log.info('Unable to read file: %s', filename)
-        return None
+        log.info('Unable to read file: %s', fjson_filename)
+        raise
     data = json.loads(file_contents)
     sections = []
+    path = ''
     title = ''
     body_content = ''
+
     if 'current_page_name' in data:
         path = data['current_page_name']
     else:
         log.info('Unable to index file due to no name %s', filename)
-        return None
```

Comment:
If we return none from this function, `get_processed_json` is going to return None too, we always expect a dict from here. And `get_processed_json` has a default https://github.com/stsewd/readthedocs.org/blob/a2e0e3f5e442b0072a4b7cbac9efadbe2a41c224/readthedocs/projects/models.py#L1254-L1261